### ISSUE TITLE
Cannot handle tapping notification event in iOS

### ISSUE THREAD

**Issue by lethanhclub** [ConnectyCube/connectycube-flutter-samples#253]
I can catch the event in android via onDidReceiveNotificationResponse and getNotificationAppLaunchDetails from flutter_local_notifications, but I cannot do that in iOS even though I am using FirebaseMessaging.onMessageOpenedApp and FirebaseMessaging.instance.getInitialMessage()

so I cannot redirect to the dialog screen when users click on push notification in iOS.

**Specify the sample to which the issue belongs (use [1]):**
[] Chat sample
[] P2P Call sample
[] Conference Call sample

**Platform (use [2])**
[] Android
[] iOS
[] macOS
[] Windows
[] Web

**Device info**
Manufacture: 
Model: 
OS version:

**Describe the bug:**

**Steps to Reproduce:**
1. 
2.
3.
4.

**Logs:**
```
log here
```
or
[Gist log](https://gist.github.com/)

**Actual result:**

**Expected behavior:**

**Additional info**

**Comment by TatankaConCube** [comment]
As you know, ConnectyCube uses APNS for working with push notifications on the iOS platform. The plugin `flutter_local_notifications` can listen to the events from the notifications published by it only. And the plugin `firebase_messaging` can only work with the FCM push notifications. The solution here will be to use a separate plugin for working with APNS. We checked the [`flutter_apns_only`](https://pub.dev/packages/flutter_apns_only) plugin and their callbacks work well with the ConnectyCube platform.

For using this plugin just follow the next steps:
1. Add the dependency `flutter_apns_only: ^1.6.0` to your `pubspec.yaml` file.
2. Init the plugin and set callbacks in your code. We did it in the `PushNotificationsManager`'s `init()` method:
```dart
if (Platform.isIOS) {
    final connector = ApnsPushConnectorOnly();

    // onLaunch gets called, when you tap on notification on a closed app
    // onResume gets called, when you tap on notification with app in background
    // onMessage gets called, when a new notification is received

    connector.configureApns(
        onLaunch: (message) async {
            print('Notification launched app, payload: ${message.payload}');
        },
        onResume: (message) async {
          print('Notification tapped when in the background, payload: ${message.payload}');
        },
        onMessage: (message) async {
          print('New notification received, payload: ${message.payload}');
        },
     );
}
```
3. Don't forget to add the next config to the `Info.plist` file if you use other Firebase services:
```xml
<key>flutter_apns.disable_swizzling</key>
<true/>
```

**Comment by lethanhclub** [comment]
Thank you very much!

I will test today

**Comment by lethanhclub** [comment]
I did 3 steps like your example, but onLaunch and onResume are not working (FirebaseMessaging.onMessageOpenedApp and FirebaseMessaging.instance.getInitialMessage() not working also).

However, onMessage (connector.configureApns) and FirebaseMessaging.onMessage are working when the app is in foreground

**Comment by TatankaConCube** [comment]
yesterday I checked it and I am sure that the callbacks `onLaunch` and `onResume` worked on my side. Please try to research on your side. 
Do you have the next code in your `AppDelegate.swift` ?
```dart
    if #available(iOS 10.0, *) {
      UNUserNotificationCenter.current().delegate = self as? UNUserNotificationCenterDelegate
    }
```

**Comment by lethanhclub** [comment]
> ```dart
> UNUserNotificationCenterDelegate
> ```

Yes, it's working if I add this code.

But I have a new problem with FirebaseMessaging.onMessageOpenedApp in iOS. These push notification I sent via firebase_messaging to my iOS devices cannot be catched by this FirebaseMessaging.onMessageOpenedApp anymore

**Comment by TatankaConCube** [comment]
Hmm... was it work before using the `flutter_apns_only`? You can use the callbacks  `onLaunch` and `onResume` for saving the id of the dialog you want to open after starting the app and after successfully login into the chat to open it.
You can get the dialog id in the callback using the next code:
```dart
          var params = message.payload;
          var dialogId = params['data']['dialog_id'];
```

**Comment by lethanhclub** [comment]
> Hmm... was it work before using the `flutter_apns_only`? You can use the callbacks `onLaunch` and `onResume` for saving the id of the dialog you want to open after starting the app and after successfully login into the chat to open it. You can get the dialog id in the callback using the next code:
> 
> ```dart
>           var params = message.payload;
>           var dialogId = params['data']['dialog_id'];
> ```

Yes FirebaseMessaging.onMessageOpenedApp was working well for iOS before I added flutter_apns_only

**Comment by TatankaConCube** [comment]
I provided the solution, try to use it or use `FirebaseMessaging.onMessageOpenedApp` without `flutter_apns_only`.

**Comment by jostney** [comment]
@lethanhclub 

Nice to see someone having similar issues. 

Are you able to NAVIGATE to dialog screen(or any you want) via notification tap on iOS ?

**Comment by jostney** [comment]
@TatankaConCube 

Wondering that notification taps not work for iOS on this sample repo normally ?

**Comment by TatankaConCube** [comment]
@jostney we pushed the solution to the separated branch [navigation_to_chat_by_notification](https://github.com/ConnectyCube/connectycube-flutter-samples/tree/fix/navigation_to_chat_by_notification) and will merge it to the master branch after detailed tests. Please check it.

**Comment by jostney** [comment]
@TatankaConCube 

I just tried the `ApnsPushConnectorOnly` on my side(not this repo) and completed notification taps for iOS successfully.

- Notification tap when app is closed 👍 
- Notification tap when app is paused 👍 

I observed trigger moment of `onLaunch` and also what happens after it. 

Let assume the app is killed or not working currently now, and we received a notification then we tapped it;

- `onLaunch` NOT triggered for now
- Our app starts
- After this`connector.configureApns(onLaunch:{...})` then `onLaunch` triggers. It seems `connector` knows the previous notification tap and it's data.


I didn’t try this repo but reviewed. 

```
PushNotificationsManager.instance.init(); 

// This line doesn’t guarantee
// 1 - Initialization of connector
// 2 - Trigger of onLaunch and storage of dialogId inside it
// Because inside of init is combination of async and sync operations.
_loginToCubeChat(context, user);
```

Instead i implemented RX way to handle navigation properly. So with this, even `onLaunch` triggers after of before the whole process completed, then the navigation works

```
  final navigatorSubject = BehaviorSubject<Map<String, dynamic>>();

  handleNotificationNavigation(ApnsRemoteMessage message) {
    try {
      navigatorSubject.add(Map<String, dynamic>.from(message.payload['data']));
    } catch (e) {}
  }

  Future<String?> _initIOS() async {
    Completer<String> completer = Completer();
    _apnsPushConnectorOnly = ApnsPushConnectorOnly();
    _apnsPushConnectorOnly?.token.addListener(() {
      completer.complete(_apnsPushConnectorOnly?.token.value);
    });
    _apnsPushConnectorOnly?.configureApns(
      onLaunch: (message) async {
        handleNotificationNavigation(message);
      },
      onResume: (message) async {
        handleNotificationNavigation(message);
      },
      onMessage: (message) async {},
    );

    _apnsPushConnectorOnly?.requestNotificationPermissions();

    return completer.future;
  }
```

**Comment by TatankaConCube** [comment]
@jostney could you please provide the links on the code you said? In our sample, we perform the login to the chat and then check if it is possible to navigate to another screen. 

What problem do you have with navigation for now? Do you need any assistance here?

**Comment by TatankaConCube** [comment]
the code from the `init` method performs faster than the `_loginToCubeChat` every time and we navigate to the required screen without any problem. Sure, some users can add some synchronous code to the `init` method and it can break the main app starting flow, but it is the developers' responsibility what do these changes.

**Comment by TatankaConCube** [comment]
fixed in version 0.7.0
